### PR TITLE
Change test_start_instance so it checks with sapcontrol

### PR DESCRIPTION
**Pull request by alvarocarvajald**
The method `test_start_instance` form `lib/sles4sap` checked for the SAP instance startup with a `ps` command. With this PR, it will check using a combination of sap commands:

- `sapcontrol -nr $instance -function GetSystemInstanceList`
- `sapcontrol -nr $instance -function GetProcessList`

This PR also introduces a time to wait for the instance to start (`WAIT_INSTANCE_START`) so as to avoid false negatives on slower machines.

- Related ticket: N/A
- Needles: N/A
- Verification run: http://mango.suse.de/tests/1289 (NetWeaver), http://mango.suse.de/tests/1290 (Hana)

### CODE REVIEW

File: lib/sles4sap.pm [hunk 1]
```diff
@@ -284,9 +284,32 @@ sub test_start_instance {
...
+
+        if ($output =~ /GREEN/) {
+            $output = script_output "sapcontrol -nr $instance -function GetProcessList | egrep -i ^[a-z]", proceed_on_failure => 1;
+            die "sapcontrol: GetProcessList: command failed" unless ($output =~ /GetProcessList[\r\n]+OK/);
+
+            my $failing_services = 0;
+            for my $line (split(/\n/, $output)) {
+                next if ($line =~ /GetProcessList|OK|^name/);
+                $failing_services++;
+                $failing_services-- if ($line =~ /GREEN/);
```
Comment (ricardobranco777) [inline lib/sles4sap.pm]:
```suggestion:-1
                $failing_services++ if ($line !~ /GREEN/);
```

Comment (ricardobranco777) [reply lib/sles4sap.pm]: Maybe remove the line above and change this one?

Comment (ricardobranco777) [inline lib/sles4sap.pm]:
```suggestion:-1
                $failing_services++ if ($line !~ /GREEN/);
```
Maybe remove the line above and use this check?

File: lib/sles4sap.pm [hunk 2]
```diff
@@ -284,9 +284,32 @@ sub test_start_instance {
...
+        die "sapcontrol: GetSystemInstanceList: command failed" unless ($output =~ /GetSystemInstanceList[\r\n]+OK/);
+
+        if ($output =~ /GREEN/) {
+            $output = script_output "sapcontrol -nr $instance -function GetProcessList | egrep -i ^[a-z]", proceed_on_failure => 1;
+            die "sapcontrol: GetProcessList: command failed" unless ($output =~ /GetProcessList[\r\n]+OK/);
+
+            my $failing_services = 0;
+            for my $line (split(/\n/, $output)) {
+                next if ($line =~ /GetProcessList|OK|^name/);
+                $failing_services++;
```
Comment (ricardobranco777) [inline lib/sles4sap.pm]:
```suggestion
```

File: lib/sles4sap.pm (hunk 1 above)
Comment (ricardobranco777) [inline lib/sles4sap.pm]:
```suggestion
                $failing_services++ if ($line !~ /GREEN/);
```

Comment (alvarocarvajald) [reply lib/sles4sap.pm]: You're correct. Will commit and test.

Comment (alvarocarvajald) [reply lib/sles4sap.pm]: New verification run: http://mango.suse.de/tests/1293

Comment (alvarocarvajald) [reply lib/sles4sap.pm]: Pushed changes. @ricardobranco777 please review again

**Comment by alvarocarvajald** [comment]
Pushed suggestions and fixed tidy issue. You can review again @ricardobranco777

**Pull request merged**